### PR TITLE
Upgrade streamlit

### PR DESCRIPTION
**Pull request by furkanmtorun**
In this PR --> still addressing #4 ,

- Upgraded streamlit and watchdog packages
- Bump the version to 1.4
- Change UI texts
- Remove footer (bug report moved to the top)
- Tab introduced for the top
- Summary text download as txt option
- UI improved
    - ROC and PR curve are now in separate expanders. 
    - Plot size is getting smaller because it was exceeding the containers.
    Screenshot 2023-03-22 at 22 14 04
- Exclude features file upload section is now changed so the CSV file does not need to include a header/title row.

Note: I plan to merge the PRs to the `OmicLearn-v1.4` branch since I realized that Cloud Streamlit is directly triggered from main.
Note 2: This is the end of the PR. I will continue working on improvements on the next PRs after merging.

### CODE REVIEW

File: omiclearn/omiclearn.py [hunk 1]
```diff
@@ -128,27 +127,31 @@ def checkpoint_for_data_upload(state, record_widgets):
...
             state["class_0"] = multiselect(
-                "Select Class 0:", unique_elements_lst, default=None
+                "Select Positive Class (e.g., Diseased/Cancer):",
+                unique_elements_lst,
+                default=None,
+                help="Select the experiment group like cancer, diseased or drug-applied group.",
             )
             state["class_1"] = multiselect(
-                "Select Class 1:",
+                "Select Negative Class (e.g., Control/Healthy):",
```
Comment (straussmaximilian) [inline omiclearn/omiclearn.py:151]: Class 1 is typically positive, https://stackoverflow.com/questions/53748375/which-class-label-is-considered-negative-in-sklearn-metrics-confusion-matrix
Maybe you can check if this is consistent across the tool.

File: docs/source/VERSION-HISTORY.md [hunk 2]
```diff
@@ -2,11 +2,28 @@
...
+> This is the latest release of **OmicLearn**.
+>
+> **Updates in this release:**
+>
+> - [x] `New Feature`: The section "Summary text" now has "Download as *.txt" button so that you can get it via a single click to save for your paper.
+> - [x] `New Feature`: The section "Additional features" supports file upload.
+> - [x] `Change`: The files to be uploaded in the section "Exclude features" now does not need to include header/title row at the top of the file. [See details here.](./USING.md)
+> - [X] The Streamlit and other packages/libraries are upgraded to their new versions.
+> - [x] Several improvements on UI and UX sides have been made.
+> - [x] OmicLearn is now faster!
```
Comment (straussmaximilian) [inline docs/source/VERSION-HISTORY.md:19]: Why is it now faster?

**Review by straussmaximilian — APPROVED** [review]
Looks good, but please double-check the convention about positive and negative class.

File: docs/source/VERSION-HISTORY.md (hunk 2 above)
Comment (furkanmtorun) [reply docs/source/VERSION-HISTORY.md:19]: - It seems they improved the `cache` strategy: https://docs.streamlit.io/library/advanced-features/caching#deciding-which-caching-decorator-to-use. 
- Also, general UI-loading performance seems improved.

**Comment by furkanmtorun** [comment]
Hi @straussmaximilian,

Thanks for raising this point! 🙏🏻 
I checked the issue, and here are my findings. I look forward to hearing your feedback on this or if you have different idea to test, just let me know! 👍🏻 

1) We do not use sklearn confusion matrix, we have our own: https://github.com/MannLabs/OmicLearn/blob/Upgrade_streamlit/omiclearn/utils/ml_helper.py#L457-L476

2) I also tried to check the code and at least, it looked good to me. 

In `omiclearn.py`, we have:
```python
subset = state.df_sub[
            state.df_sub[state.target_column].isin(state.class_0)
            | state.df_sub[state.target_column].isin(state.class_1)
].copy()		
state.y = subset[state.target_column].isin(state.class_0)
```
So, here, within `state.y` pd.Series, we turn the `class_0` ones to `True` (positive label) and `class_1` ones become `False`. Then, as you can see in `calculate_cm` in the answer point 1, we calculate `True Positive` with: 
  ```python
  if y_test[i] == y_pred[i] == True:
	  tp += 1
  ``` 

Here, in the example, I set the positive class as "a" and negative as "b" (as you can also see from the terminal SS) and I printed `y_test` within `elif metric_name in ["precision", "recall", "f1"]:` condition to test and I compared it with the DF (based on row-index number):
df
y_test
y_test_2



3) Also, here is another check for checking labels. Please, see following 2 screenshots:

a-Neg__b-Pos
a-Pos___b-Neg

**Comment by straussmaximilian** [comment]
> Hi @straussmaximilian,
> 
> Thanks for raising this point! 🙏🏻 I checked the issue, and here are my findings. I look forward to hearing your feedback on this or if you have different idea to test, just let me know! 👍🏻
> 
> 1. We do not use sklearn confusion matrix, we have our own: https://github.com/MannLabs/OmicLearn/blob/Upgrade_streamlit/omiclearn/utils/ml_helper.py#L457-L476
> 2. I also tried to check the code and at least, it looked good to me.
> 
> In `omiclearn.py`, we have:
> 
> ```python
> subset = state.df_sub[
>             state.df_sub[state.target_column].isin(state.class_0)
>             | state.df_sub[state.target_column].isin(state.class_1)
> ].copy()		
> state.y = subset[state.target_column].isin(state.class_0)
> ```
> 
> So, here, within `state.y` pd.Series, we turn the `class_0` ones to `True` (positive label) and `class_1` ones become `False`. Then, as you can see in `calculate_cm` in the answer point 1, we calculate `True Positive` with:
> 
> ```python
> if y_test[i] == y_pred[i] == True:
>     tp += 1
> ```
> 
> Here, in the example, I set the positive class as "a" and negative as "b" (as you can also see from the terminal SS) and I printed `y_test` within `elif metric_name in ["precision", "recall", "f1"]:` condition to test and I compared it with the DF (based on row-index number): df y_test y_test_2
> 
> 3. Also, here is another check for checking labels. Please, see following 2 screenshots:
> 
> a-Neg__b-Pos a-Pos___b-Neg

This makes sense to me. Looks consistent then! Thanks for checking.

**Pull request merged**